src/main/java/com/teragrep/pth_06/planner/BloomFilterTempTable.java

```diff
@@ -0,0 +1,231 @@
+        }
+        ctx.dropTemporaryTableIfExists(table).execute();
+        final String sql = "create temporary table " + table.getName()
+                + "(id bigint auto_increment primary key, term_id bigint, type_id bigint, filter longblob, unique key "
+                + table.getName() + "_unique_key (term_id, type_id))";
+        final Query query = ctx.query(sql);
+        query.execute();
+        final Index logtimeIndex = DSL.index(DSL.name(table.getName() + "_ix_type_id"));
+        final CreateIndexIncludeStep createIndexIncludeStep = ctx.createIndex(logtimeIndex).on(table, typeIdField);
+        LOGGER.trace("BloomFilterTempTable create index <{}>", createIndexIncludeStep);
+        createIndexIncludeStep.execute();
+    }
+
+    /**
+     * Inserts a filter for each filtertype inside parent table records. Filter is filled with search term token set,
+     * filter size is selected using parent table filtertype expected and fpp values.
+     */
+    private void insertFilters() {
+        final Table<?> joined = parentTable
+                .join(BLOOMDB.FILTERTYPE)
+                .on(BLOOMDB.FILTERTYPE.ID.eq((Field<ULong>) parentTable.field("filter_type_id")));
+        final Field<ULong> expectedField = DSL.field(DSL.name(table.getName(), "expectedElements"), ULong.class);
+        final Field<Double> fppField = DSL.field(DSL.name(table.getName(), "targetFpp"), Double.class);
+        final SelectField<?>[] resultFields = {
+                BLOOMDB.FILTERTYPE.ID,
+                joined.field("expectedElements").as(expectedField),
+                joined.field("targetFpp").as(fppField)
+        };
+        // Fetch filtertype values
+        final Result<Record> records = ctx
+                .select(resultFields)
+                .from(joined)
+                .groupBy(joined.field("filter_type_id"))
+                .fetch();
+        if (records.isEmpty()) {
+            throw new RuntimeException("Parent table was empty");
+        }
+        for (Record record : records) {
+            insertFilterFromRecord(record, expectedField, fppField);
+        }
+    }
+
+    private void insertFilterFromRecord(
+            final Record record,
+            final Field<ULong> expectedField,
+            final Field<Double> fppField
+    ) {
+        final ULong filterTypeId = record.getValue(BLOOMDB.FILTERTYPE.ID); // filter_type_id
+        final ULong expected = record.getValue(expectedField); // expectedElements
+        final Double fpp = record.getValue(fppField); // targetFpp
+        final BloomFilter filter = BloomFilter.create(expected.longValue(), fpp);
+        tokenSet.forEach(token -> filter.put(token.toString()));
+        final ByteArrayOutputStream filterBAOS = new ByteArrayOutputStream();
+        try {
+            filter.writeTo(filterBAOS);
+            filterBAOS.close();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
```

Comment:
perhaps it would be worth it to add some extra context to this exception in addition to the original exception to help debugging in the future.